lib/SPIRVProducerPass.cpp

```diff
@@ -1661,19 +1661,20 @@ SPIRVID SPIRVProducerPass::getSPIRVType(Type *Ty) {
         RID = addSPIRVInst<kTypes>(spv::OpTypeSampler);
         break;
       } else if (STy->getName().startswith("opencl.image1d_ro_t") ||
+                 STy->getName().startswith("opencl.image1d_rw_t") ||
                  STy->getName().startswith("opencl.image1d_wo_t") ||
                  STy->getName().startswith("opencl.image1d_array_ro_t") ||
+                 STy->getName().startswith("opencl.image1d_array_rw_t") ||
                  STy->getName().startswith("opencl.image1d_array_wo_t") ||
                  STy->getName().startswith("opencl.image2d_ro_t") ||
+                 STy->getName().startswith("opencl.image2d_rw_t") ||
                  STy->getName().startswith("opencl.image2d_wo_t") ||
                  STy->getName().startswith("opencl.image2d_array_ro_t") ||
+                 STy->getName().startswith("opencl.image2d_array_rw_t") ||
                  STy->getName().startswith("opencl.image2d_array_wo_t") ||
                  STy->getName().startswith("opencl.image3d_ro_t") ||
+                 STy->getName().startswith("opencl.image3d_rw_t") ||
                  STy->getName().startswith("opencl.image3d_wo_t")) {
-
-        if (STy->getName().contains("_wo_t")) {
-          addCapability(spv::CapabilityStorageImageWriteWithoutFormat);
```

Comment:
Ah, you're just making this more specialized to the cases where it's needed.
Sounds good.